driver-core/src/main/com/mongodb/internal/Exceptions.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.internal;
+
+import com.mongodb.MongoCommandException;
+import org.bson.BsonArray;
+import org.bson.BsonDocument;
+import org.bson.BsonInt32;
+import org.bson.BsonNumber;
+import org.bson.BsonString;
+import org.bson.BsonValue;
+
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.mongodb.internal.VisibleForTesting.AccessModifier.PRIVATE;
+
+/**
+ * <p>This class is not part of the public API and may be removed or changed at any time</p>
+ */
+public final class Exceptions {
+    public static final class MongoCommandExceptions {
```

Comment:
> I've never seen this particular nested class pattern before.

It's something we started doing in the Kafka connector:

- https://github.com/mongodb/mongo-kafka/pull/121/files?show-viewed-files=true&file-filters%5B%5D=#r1003570842
- https://github.com/mongodb/mongo-kafka/blob/master/src/test/java/com/mongodb/kafka/connect/source/MongoSourceConfigTest.java#L549,

Only at that time I didn't know one must use `@Nested`, but now IntelliJ warned me and I learned.

> Is it in anticipation of adding more nested classes in the future for other exception types?

Yes. Nesting is one more tool for grouping tests: https://stackoverflow.com/questions/36220889/whats-the-purpose-of-the-junit-5-nested-annotation.

Reply:
> P.S. I'll need to add @Nested to the Kafka connector codebase.

Done: https://github.com/mongodb/mongo-kafka/commit/62fedf0da3e956d6fcd2864614a81cf0637f9b2d

Reply:
The test classes are nested because the classes they test are. The reasons the tested classes are nested are:

- I want utilities for specific exception types to be separated from utilities for any exception.
- On the other hand, I still want utilities for specific exception types to be somehow grouped together, and grouped with utilities for any exception.

I see two ways of achieving the above:

1. Use nested classes, as currently done in the PR.
2. Use a new `com.mongodb.internal.exception` package.

I chose 1, but I am not opposed to 2. Let me know if you think we should do 2, and I'll refactor the code and the test.

Reply:
Thank you for the clarification. I don't have a strong preference, since nesting or packaging serve the same purpose here, as you mentioned. Feel free to decide as you see fit.